### PR TITLE
Add Bitfinex Spot exchange

### PR DESCRIPTION
**Pull request by artememelin**
### Pull request checklist

- [x] Did you bump the project version?
- [x] Did you add a description to the Pull Request?
- [ ] Did you add new tests?
- [x] Did you add reviewers?

### CODE REVIEW

File: src/Bitfinex/Spot/API/Candle.jl [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+
+using CryptoAPIs.Bitfinex
+using CryptoAPIs: Maybe, APIsRequest
+
+@enum TimeInterval m1 m5 m15 m30 h1 h3 h6 h12 d1 w1 d14 M1
+
+Base.@kwdef struct CandleQuery <: BitfinexPublicQuery
+    timeframe::TimeInterval
+    symbol::String
+    section::String = "hist"
```
Comment (dmitrii-doronin) [inline src/Bitfinex/Spot/API/Candle.jl:18]: Is it a constant? Are there other options?

Comment (dmitrii-doronin) [reply src/Bitfinex/Spot/API/Candle.jl:18]: How about an enum for that?

File: src/Bitfinex/Bitfinex.jl [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+"""
+    BitfinexAPIError{T} <: AbstractAPIsError
+
+Exception thrown when an API method fails with code `T`.
+
+## Required fields
+- `type::String`: Error type.
+- `code::Int64`: Error code.
+- `msg::String`: Error message.
```
Comment (yshadrin) [inline src/Bitfinex/Bitfinex.jl:57]: Аre these fields required or optional?

File: src/Bitfinex/Bitfinex.jl [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+end
+
+CryptoAPIs.error_type(::BitfinexClient) = BitfinexAPIError
+
+function CryptoAPIs.request_sign!(::BitfinexClient, query::Q, ::String)::Q where {Q<:BitfinexPublicQuery}
+    return query
+end
+
+function CryptoAPIs.request_sign!(client::BitfinexClient, query::Q, endpoint::String)::Nothing where {Q<:BitfinexPrivateQuery}
+    query.nonce = string(time2unix(now(UTC)))
```
Comment (yshadrin) [inline src/Bitfinex/Bitfinex.jl:80]: what is the function `time2unix`?

File: src/Bitfinex/Spot/API/TradesPair.jl [hunk 4]
```diff
@@ -0,0 +1,84 @@
+module TradesPair
```
Comment (yshadrin) [inline src/Bitfinex/Spot/API/TradesPair.jl:1]: let's name the module as `TradePair`

File: examples/Bitfinex/Spot.jl [hunk 5]
```diff
@@ -0,0 +1,27 @@
+# Bitfinex/Spot
+# https://docs.bitfinex.com/docs/rest-public
+
+using Dates
+using CryptoAPIs
+using CryptoAPIs.Bitfinex
+
+
```
Comment (yshadrin) [inline examples/Bitfinex/Spot.jl:8]: extra enter

File: src/Bitfinex/Spot/API/OrderBook.jl [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+
+[`GET v2/book/{symbol}/{precision}`](https://docs.bitfinex.com/reference/rest-public-book)
+
+## Parameters:
+
+| Parameter | Type            | Required | Description |
+|:----------|:----------------|:---------|:------------|
+| symbol    | String          | true     |             |
+| precision | Precision       | true     | `P0` `P1` `P2` `P3` `P4` `R0` |
+| len       | OrderBookLength | false    | Default: `ONE` (1), Available: `TWENTY_FIVE` (25), `ONE_HUNDRED` (100). |
```
Comment (yshadrin) [inline src/Bitfinex/Spot/API/OrderBook.jl:55]: fix formatting

File: src/Bitfinex/Spot/API/Candle.jl [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+    "low":67032.0,
+    "volume":0.32812637
+  },
+  ...
+]
+```
+"""
+function candle(client::BitfinexClient, query::CandleQuery)
+    timeframe = Serde.ser_type(CandleQuery, query.timeframe)
+    section   = Serde.ser_type(CandleQuery, query.section)
```
Comment (yshadrin) [inline src/Bitfinex/Spot/API/Candle.jl:108]:
```suggestion
    section = Serde.ser_type(CandleQuery, query.section)
```

**Review by yshadrin — APPROVED** [review]

**Review by gryumov — APPROVED** [review]

**Pull request merged**